cpp/include/cuml/svm/svm_model.h

```diff
@@ -24,22 +28,111 @@ namespace SVM {
  */
 template <typename math_t>
 struct svmModel {
```

Comment:
For SVMs we do not know the number of model parameters (dual_coef_, support_, support_vector_) until the model is fitted. Before this PR, the svmModel structure just stored raw pointers which were allocated and deallocated by the c++ svm routines.

You have introduced the rmm::device_uvector fields in svmModel (device_uvector has cython wrappers). After training, these are resized to store the model parameters. This achieves the goal of removing references of raft::allocator.

This change also means that svmModel is not a POD struct. Our developer guide requires that we should have only POD args for the API. 

To keep svmModel POD, I see the following solutions:
1. keep the device alloc/dealloc calls
2. Let svmFit return a fixed size (nrows, 1) arrays of dual_coef_ and support_. We can post process these to omit the elements with zero dual coefs, and also collect the corresponding support vector into the array support_vector_ (n_sv, n_cols). Effectively this would mean thet Results::Get needs a cython wrapper, and we would call this from svm_base.pyx.
3. Have a C compatible handle for rmm_device_uvector, and store these in svmModel.

Reply:
We finally opted for the first solution.